homeassistant/components/tuya/vacuum.py

```diff
@@ -78,51 +78,53 @@ class TuyaVacuumEntity(TuyaEntity, StateVacuumEntity):
 
     _fan_speed: EnumTypeData | None = None
     _battery_level: IntegerTypeData | None = None
-    _supported_features = 0
 
     def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager) -> None:
         """Init Tuya vacuum."""
         super().__init__(device, device_manager)
 
-        self._supported_features |= VacuumEntityFeature.SEND_COMMAND
+        self._attr_supported_features = 0
+        self._attr_fan_speed_list = []
+
+        self._attr_supported_features |= VacuumEntityFeature.SEND_COMMAND
         if self.find_dpcode(DPCode.PAUSE, prefer_function=True):
-            self._supported_features |= VacuumEntityFeature.PAUSE
+            self._attr_supported_features |= VacuumEntityFeature.PAUSE
 
         if self.find_dpcode(DPCode.SWITCH_CHARGE, prefer_function=True):
-            self._supported_features |= VacuumEntityFeature.RETURN_HOME
+            self._attr_supported_features |= VacuumEntityFeature.RETURN_HOME
         elif (
             enum_type := self.find_dpcode(
                 DPCode.MODE, dptype=DPType.ENUM, prefer_function=True
             )
         ) and TUYA_MODE_RETURN_HOME in enum_type.range:
-            self._supported_features |= VacuumEntityFeature.RETURN_HOME
+            self._attr_supported_features |= VacuumEntityFeature.RETURN_HOME
 
         if self.find_dpcode(DPCode.SEEK, prefer_function=True):
-            self._supported_features |= VacuumEntityFeature.LOCATE
+            self._attr_supported_features |= VacuumEntityFeature.LOCATE
 
         if self.find_dpcode(DPCode.STATUS, prefer_function=True):
-            self._supported_features |= (
+            self._attr_supported_features |= (
                 VacuumEntityFeature.STATE | VacuumEntityFeature.STATUS
             )
 
         if self.find_dpcode(DPCode.POWER, prefer_function=True):
-            self._supported_features |= (
+            self._attr_supported_features |= (
                 VacuumEntityFeature.TURN_ON | VacuumEntityFeature.TURN_OFF
             )
 
         if self.find_dpcode(DPCode.POWER_GO, prefer_function=True):
-            self._supported_features |= (
+            self._attr_supported_features |= (
                 VacuumEntityFeature.STOP | VacuumEntityFeature.START
             )
 
         if enum_type := self.find_dpcode(
             DPCode.SUCTION, dptype=DPType.ENUM, prefer_function=True
         ):
-            self._supported_features |= VacuumEntityFeature.FAN_SPEED
-            self._fan_speed = enum_type
+            self._attr_supported_features |= VacuumEntityFeature.FAN_SPEED
+            self._attr_fan_speed_list = enum_type.range
```

Comment:
This is actually a bug. The enum_type for fan speed needs to be stored (and used). The issue is; the bug is: It isn't used :)

Reply:
I've reinstated the _fan_speed attribute, but I'm not sure I can help with the bug that still is...